extension/src/lambda_request_error.rs

```diff
@@ -64,53 +64,22 @@ impl From<&LambdaRequestError> for ExitReason {
 
 impl LambdaRequestError {
   pub fn is_fatal(&self) -> bool {
-    use LambdaRequestError::*;
-
-    let fatal_errors = [
-      // Add other fatal errors here
-      AppConnectionUnreachable,
-    ];
-
-    fatal_errors.contains(self)
-  }
-
-  pub fn worse(self, other: Self) -> Self {
-    use LambdaRequestError::*;
-
-    let ordered_reasons = [
-      RouterLambdaInvokeInvalid,
-      RouterUnreachable,
-      RouterConnectionError,
-      AppConnectionUnreachable,
-      AppConnectionError,
-      ChannelErrorOther,
-    ];
-
-    for reason in ordered_reasons {
-      if self == reason || other == reason {
-        return reason;
-      }
-    }
-
-    self // If no match is found, return the current value
+    // Add other fatal errors to the match pattern. For example:
+    // matches!(self, Self::Variant1 | Self::Variant2)
+    matches!(self, Self::AppConnectionUnreachable)
   }
+}
 
-  // This function will cause a compile-time error if a new variant is added to the enum
-  // but not added to the match expression.
-  #[allow(dead_code)]
-  fn ensure_all_variants_handled(variant: Self) {
-    use LambdaRequestError::*;
```

Comment:
Wildcard using the variants in an enum can be a gotcha. If someone to remove a variant from the enum, the following match statement wouldn't catch it. Instead the variant that was removed would become a variable declaration for a catchall branch of the match. Like using a `_`.